core/src/test/scala/integration/kafka/admin/AdminFenceProducersIntegrationTest.scala

```diff
@@ -107,6 +110,43 @@ class AdminFenceProducersIntegrationTest extends IntegrationTestHarness {
     assertThrows(classOf[ProducerFencedException], () => producer.commitTransaction())
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = Array("zk", "kraft"))
+  @Timeout(value = 30)
+  def testFenceProducerTimeoutMs(quorum: String): Unit = {
```

Comment:
this test case is used to verify the timeout, so we don't need to send transaction actually. 

```scala
  @ParameterizedTest
  @ValueSource(strings = Array("zk", "kraft"))
  @Timeout(value = 30)
  def testFenceProducerTimeoutMs(quorum: String): Unit = {
    val config = createConfig
    config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, s"localhost:${TestUtils.IncorrectBrokerPort}")
    val admin = Admin.create(config)
    try {
      val e = assertThrows(classOf[ExecutionException], () => admin.fenceProducers(Collections.singletonList(txnId),
        new FenceProducersOptions().timeoutMs(0)).all().get())
      assertInstanceOf(classOf[TimeoutException], e.getCause)
    } finally admin.close(java.time.Duration.ofSeconds(0))
  }
```